kv/kvserver: TestGossipHandlesReplacedNode failed

kv/kvserver.TestGossipHandlesReplacedNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/c5eaf357-4b6f-405a-9f5d-2cc47da1c77f?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [485975b3a824c68c07340e6a336c7864c00d3c6d](https://github.com/cockroachdb/cockroach/commits/485975b3a824c68c07340e6a336c7864c00d3c6d):


```
=== RUN   TestGossipHandlesReplacedNode
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestGossipHandlesReplacedNode572517143
    test_log_scope.go:81: use -show-logs to present logs inline
    testcluster.go:687: timed out waiting for 4 store descriptors: map[2:{} 3:{} 4:{}]
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestGossipHandlesReplacedNode572517143
--- FAIL: TestGossipHandlesReplacedNode (225.46s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126790 kv/kvserver: TestGossipHandlesReplacedNode failed [C-test-failure O-robot T-kv branch-release-24.1.2-rc release-blocker]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGossipHandlesReplacedNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40152

--- a/pkg/kv/kvserver/gossip_test.go
+++ b/pkg/kv/kvserver/gossip_test.go
@@ -147,6 +147,7 @@ func TestGossipHandlesReplacedNode(t *testing.T) {
 	// As of Nov 2018 it takes 3.6s.
 	skip.UnderShort(t)
 	skip.UnderDeadlock(t)
+	skip.UnderRace(t)
 	ctx := context.Background()
 
 	// Shorten the raft tick interval and election timeout to make range leases

